src/audits/ImageWithoutAltText.js

```diff
@@ -27,7 +28,16 @@ axs.AuditRules.addRule({
             !axs.utils.isElementOrAncestorHidden(element);
     },
     test: function(image) {
-        return (!image.hasAttribute('alt') && image.getAttribute('role') != 'presentation');
+        var imageIsPresentational = (image.hasAttribute('alt') && image.alt == "") || image.getAttribute('role') == 'presentation';
+        if (imageIsPresentational)
+            return false;
+        var textAlternatives = axs.properties.getTextProperties(image);
+        if (!textAlternatives)
+            return true;
+        // filename is only present if no other option found
```

Comment:
Is this comment necessary? I'm not really sure what it means, and I'm guessing it's ok to remove it for the sake of tidiness.